mephisto/client/review_app/server/api/views/units_details_view.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List
+from typing import Optional
+
+from flask import current_app as app
+from flask import request
+from flask.views import MethodView
+from werkzeug.exceptions import BadRequest
+
+from mephisto.data_model.unit import Unit
+
+
+class UnitsDetailsView(MethodView):
+    def get(self) -> dict:
+        """ Get full input for specified workers results (`unit_ids` is mandatory) """
+
+        unit_ids: Optional[str] = request.args.get("unit_ids")
+
+        app.logger.debug(f"Params: {unit_ids=}")
+
+        # Parse `unit_ids`
+        if unit_ids:
+            try:
+                unit_ids: List[int] = [int(i.strip()) for i in unit_ids.split(",")]
+            except ValueError:
+                raise BadRequest("`unit_ids` must be a comma-separated list of integers.")
+
+        # Validate params
+        if not unit_ids:
+            raise BadRequest("`unit_ids` parameter must be specified.")
+
+        # Get units
+        db_units: List[Unit] = app.db.find_units()
+
+        # Prepare response
+        units = []
+        for unit in db_units:
+            if unit_ids and int(unit.db_id) not in unit_ids:
+                continue
+
+            try:
+                unit_data = app.data_browser.get_data_from_unit(unit)
+            except AssertionError:
+                # In case if this is Expired Unit. It raises and axceptions
+                unit_data = {}
+
+            units.append(
```

Comment:
I think that, for the most part, this should end up just being `units.append(unit_data)`. Ultimately though I see what you're saying, this is going to require a significant change to how `AgentState` specifies data, at least insofar as providing separate `inputs` (which will be what Mephisto normally sends to `initialTaskData` when a task is being done, and `outputs` (which can be _anything else_ that `AgentState.get_data` wants to include in the final data). This should be standardized across Mephisto, which unfortunately will incur us setting up a migration (either a util for going from the old formats to the standard, or temporarily dealing with both).

Then `TaskFrontend` can by default accept `initialTaskData` and `taskOutputs` or something.